application/src/main/java/io/vertx/core/application/impl/VertxApplicationCommand.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright (c) 2011-2023 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.application.impl;
+
+import io.vertx.core.*;
+import io.vertx.core.application.ExitCodes;
+import io.vertx.core.application.VertxApplication;
+import io.vertx.core.application.VertxApplicationHooks;
+import io.vertx.core.eventbus.EventBusOptions;
+import io.vertx.core.impl.VertxBuilder;
+import io.vertx.core.impl.VertxInternal;
+import io.vertx.core.impl.logging.Logger;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.metrics.MetricsOptions;
+import io.vertx.core.spi.VertxMetricsFactory;
+import io.vertx.core.spi.VertxServiceProvider;
+import io.vertx.core.spi.VertxTracerFactory;
+import io.vertx.core.tracing.TracingOptions;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+
+import static io.vertx.core.application.impl.Utils.*;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.stream.Collectors.toList;
+import static picocli.CommandLine.Parameters.NULL_VALUE;
+
+@Command(name = "VertxApplication", description = "Runs a Vert.x application.", sortOptions = false)
+public class VertxApplicationCommand implements Runnable {
```

Comment:
I think we should make this public and allow the user subclassing it to add new options

Reply:
I would prefer that, for Vertx 5.0, we keep it internal and see if we get requests for extensibility. And then we can determine what's best and how to do it (subclassing, mixin... etc).

Reply:
that works, it's not high priority, but I think there is definitely value here